### DISCUSSION TITLE
RemoteDebuggingPort remote-allow-origins

### DISCUSSION THREAD

**Discussion by Zer0bytte** [cefsharp/CefSharp#4470] [General]
![image](https://user-images.githubusercontent.com/85482481/232208755-f3468673-0532-4350-99e0-f02c2f5f3371.png)
![image](https://user-images.githubusercontent.com/85482481/232208778-5fcd13c5-3992-411f-9f2e-e1172ef13745.png)
When trying to open remote debugging this issue happens cefsharp v112 (latest version) and .net 7

**Comment by n303ef** [comment]
Try this:

`cefSettings.CefCommandLineArgs.Add("remote-allow-origins", "*");`

**Comment by amaitland** [comment]
In future please use [Discussions](https://github.com/cefsharp/CefSharp/discussions) to ask support related questions.

Additionally [please no images of code](https://github.com/cefsharp/CefSharp/wiki/Questions-&-Support#no-images-of-code), cannot copy and paste from an image.

As @n303ef has pointed out your code is incorrect. It should be:

```c#
settings.CefCommandLineArgs.Add("remote-allow-origins", "*");
```

For security reasons wherever possible only allow specific origin(s).

```c#
settings.CefCommandLineArgs.Add("remote-allow-origins", "http://localhost:9955");
```

```
[0416/184046.922:ERROR:devtools_http_handler.cc(766)] Rejected an incoming WebSocket connection from the http://localhost:9995 origin. Use the command line flag --remote-allow-origins=http://localhost:9995 to allow connections from this origin or --remote-allow-origins=* to allow all origins.
```

`Chromium` will make a suggestion on which origin to allow in the log file.

**Comment by Zer0bytte** [comment]
Thanks @amaitland  @n303ef

**Comment by markosole** [comment]
I was looking into same issue or feature for same reason. I've tried with different flags 

```c++
settings.CefCommandLineArgs.Add("remote-allow-origins", "*");
``` 

### And 
```c++
settings.CefCommandLineArgs.Add("remote-allow-origins", "http://localhost:9222");
// ...
``` 

### And with all combinations incl this as well 

```c++
settings.CefCommandLineArgs.Add("remote-debugging-port", "9222");
settings.CefCommandLineArgs.Add("remote-debugging-address", "0.0.0.0");
settings.CefCommandLineArgs.Add("disable-features", "BlockInsecurePrivateNetworkRequests");        
            
```

I still could not access it from any other machine but localhost and 127.0.0.1. Even if I set 

```c++
settings.CefCommandLineArgs.Add("remote-debugging-address", "192.168.0.211");  // for example
```
It still would not let me to connect when accessing via browser http://192.168.0.211
Have anyone had success on accessing it from other PC on the same network and what options have you used?

 - Sidenote: 
 I took another approach by using Proxy and that worked half way through. Debugging is using ws (web sockets) and that was the issue as proxy I was using would not pass `ws` through. 
 
 **CEF version I am using is: 101.0.150**

**Reply by amaitland** [reply]
> CEF version I am using is: 101.0.150

Are you sure the command line args are supported in this version? It's 11 versions behind the version mentioned in this discussion.

Try testing with the current supported version.
